controllers/propagator/policy_controller.go

```diff
@@ -160,6 +174,20 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 			return reconcile.Result{RequeueAfter: duration}, nil
 		}
 
+		// Collect the hub template watch metric; If there's an error, just log it
+		hubTempWatches := r.DynamicWatcher.GetWatchCount()
```

Comment:
Do you think we could put this code in a deferred function instead?

Reply:
Ah, that's a good idea. I'll update it.